code/modules/events/undead.dm

```diff
@@ -0,0 +1,63 @@
+
+/datum/round_event_control/undead
+	name = "Raise Dead"
+	typepath = /datum/round_event/ghost_role/undead
+	weight = 10
+
+/datum/round_event/ghost_role/undead
+	minimum_required = 1
+	role_name = "friendly zombie"
+	var/spawns = 3
+	var/one = "one"
+	fakeable = TRUE
+
+/datum/round_event/ghost_role/undead/announce(fake)
+	var/source = pick("a powerful spell from a far away wizard", "a strange anomalous enegy", "the influence of a high level necromancer", "the power of love", "the will of an omnipotent God", "\[REDACTED\]")
+	var/purpose = pick("party like there's no tomorrow", "seek employment", "partake in the democratic process", "fulfill their new year's promises", "have an after-life crysis", "preach about their new religion", "answer your questions about life, the universe, and everything", "persuade you not to treat them like their brain eating counterparts", "repair their past mistakes", "\[REDACTED\]")
+
+	priority_announce("Your station has been subjected to [source]. Some dead bodies have come back to life to [purpose]. On behalf of Nanotrasen, please welcome these former employees as your new coworkers!","[command_name()] Medium-Priority Update")
+
+/datum/round_event/ghost_role/undead/spawn_role()
+	var/list/mob/dead/observer/candidates = get_candidates(ROLE_REVENANT, null, ROLE_REVENANT)
```

Comment:
Role revenant?

Reply:
What do you suggest?

Reply:
?

Reply:
Code is from sentience potion, who uses the xenomorph role for some reason.

I think roles need a revision.

Reply:
To be fair current revenant is closer to a poltergheist than revenants.